src/main/java/de/retest/recheck/report/TestReportFilter.java

```diff
@@ -146,7 +147,12 @@ StateDifference filter( final StateDifference stateDiff ) {
 	}
 
 	Optional<InsertedDeletedElementDifference> filter( final InsertedDeletedElementDifference insertedDeletedDiff ) {
-		return filter.matches( insertedDeletedDiff.getInsertedOrDeletedElement() ) //
+		if ( insertedDeletedDiff.isInserted() ) {
+			return filter.matches( insertedDeletedDiff.getInsertedOrDeletedElement(), ChangeType.Inserted ) //
+					? Optional.empty() //
+					: Optional.of( insertedDeletedDiff );
+		}
+		return filter.matches( insertedDeletedDiff.getInsertedOrDeletedElement(), ChangeType.Deleted ) //
```

Comment:
Please add at least one test for each where you define such a filter to check if the method is properly called...